src/runtime_src/core/common/api/xrt_device.cpp

```diff
@@ -209,6 +219,25 @@ xrtDeviceLoadXclbinFile(xrtDeviceHandle dhdl, const char* fnm)
   }
 }
 
+int
+xrtDeviceLoadXclbinHandle(xrtDeviceHandle dhdl, xrtXclbinHandle xhdl)
+{
+  try {
+    auto device = get_device(dhdl);
+    auto xclbin = xrt_core::xclbin_int::get_xclbin_data(xhdl);
```

Comment:
auto& to avoid copying the vector which is returned by reference.  This is one of those cases where auto makes it a bit tricky because I am sure you'd never make this mistake if you typed std::vector<char>.   Still auto should be used.  

Alternatively, since you never really need the vector and are returned a reference, maybe you should be returning const char* from xrt_core::xclbin_int::get_xclbin_data(...) ?

Reply:
Good point Soren. We changed to auto&.